baybe/surrogates/base.py

```diff
@@ -74,45 +96,33 @@ class Surrogate(ABC, SerialMixin):
     """Class variable encoding whether or not the surrogate supports transfer
     learning."""
 
-    _input_transform: Callable[[pd.DataFrame], pd.DataFrame] | None = field(
-        init=False, default=None, eq=False
-    )
-    """Callable preparing surrogate inputs for training/prediction.
-
-    Transforms a dataframe containing parameter configurations in experimental
-    representation to a corresponding dataframe containing their **scaled**
-    computational representation. Only available after the surrogate has been fitted."""
+    _searchspace: SearchSpace | None = field(init=False, default=None, eq=False)
+    """The search space on which the surrogate operates. Available after fitting."""
 
-    _output_transform: Callable[[pd.DataFrame], pd.DataFrame] | None = field(
-        init=False, default=None, eq=False
-    )
-    """Callable preparing surrogate outputs for training.
-
-    Transforms a dataframe containing target measurements in experimental
-    representation to a corresponding dataframe containing their **scaled**
-    computational representation. Only available after the surrogate has been fitted."""
-
-    # TODO: type should be `Standardize | _NoTransform`` but is currently
-    #   omitted due to: https://github.com/python-attrs/cattrs/issues/531
+    # TODO: type should be
+    #   `botorch.models.transforms.outcome.Standardize | _NoTransform`
+    #   but is currently omitted due to:
+    #   https://github.com/python-attrs/cattrs/issues/531
     _output_scaler = field(init=False, default=None, eq=False)
-    """Optional callable for scaling output values.
+    """Scaler for transforming output values. Available after fitting.
 
     Scales a tensor containing target measurements in computational representation
-    to make them ready for processing by the surrogate. Only available after the
-    surrogate has been fitted."""
+    to make them digestible for the model-specific, scale-agnostic posterior logic."""
 
-    def to_botorch(self) -> Model:
-        """Create the botorch-ready representation of the model."""
+    def to_botorch(self) -> Model:  # noqa: D102
+        # See base class.
         from baybe.surrogates._adapter import AdapterModel
 
         return AdapterModel(self)
 
     @staticmethod
-    def _make_parameter_scaler(
+    def _make_parameter_scaler_factory(
         parameter: Parameter,
-    ) -> ParameterScalerProtocol | None:
-        """Return the scaler to be used for the given parameter."""
-        return MinMaxScaler()
+    ) -> type[InputTransform] | None:
```

Comment:
Why do we need the `type` here but not for the `OutcomeTransform`? Because it is the factory?

Reply:
Yes, exactly. Right now, the `OutcomeTransform` still returns the object itself, which might change however, depending on the outcome of [this discussion](https://github.com/emdgroup/baybe/pull/325#discussion_r1695872331). The `type` needs to stay in any case, though